packages/common/src/ProviderUtils.ts

```diff
@@ -151,6 +151,25 @@ export function getWeb3ByChainId(chainId: number): Web3 {
   return new Web3(keyedProvider);
 }
 
+/**
+ * @notice Creates array of web3 instances for a particular chain.
+ * @dev Providers to use are described in RETRY_CONFIG_{chainId} dictionary under the "url" key.
+ * @param chainId the chain id for the network the user wants to connect to.
+ * @returns array of new readonly Web3 instances.
+ */
+export function getRetryWeb3sByChainId(chainId: number): Web3[] {
+  const retryConfigJson = process.env[`RETRY_CONFIG_${chainId}`];
+  if (!retryConfigJson) return [];
+  const retryConfig: RetryConfig[] = JSON.parse(retryConfigJson);
+
+  const retryWeb3s: Web3[] = [];
+  retryConfig.forEach((config) => {
+    retryWeb3s.push(new Web3(config.url));
+  });
+
+  return retryWeb3s;
```

Comment:
I think this can be simplified to remove the special cases.

```suggestion
  const retryConfigJson = process.env[`RETRY_CONFIG_${chainId}`] || "[]";
  const retryConfig: RetryConfig[] = JSON.parse(retryConfigJson);
  return retryConfig.map((config) => new Web3(config.url));
```